src/main/java/de/hysky/skyblocker/skyblock/tabhud/widget/component/ProgressComponent.java

```diff
@@ -62,8 +65,14 @@ public void render(DrawContext context, int x, int y) {
 		int barY = y + txtRend.fontHeight + PAD_S;
 		int endOffsX = ((int) (this.barW * (this.pcnt / 100f)));
 		context.fill(barX + endOffsX, barY, barX + this.barW, barY + BAR_HEIGHT, COL_BG_BAR);
-		context.fill(barX, barY, barX + endOffsX, barY + BAR_HEIGHT,
-				this.color);
-		context.drawTextWithShadow(txtRend, bar, barX + 3, barY + 2, 0xffffffff);
+		context.fill(barX, barY, barX + endOffsX, barY + BAR_HEIGHT, this.color);
+
+		int textWidth = txtRend.getWidth(bar);
+		int textColor = 0xffffffff;
+		if (endOffsX >= textWidth) {
+			textColor = (this.colorIsBright) ? 0xff000000 : 0xffffffff;
+		}
+
+		context.drawText(txtRend, bar, barX + 3, barY + 2, textColor, textColor != 0xff000000);
```

Comment:
```suggestion
		// Only turn text dark when it is wider than the filled bar and the filled bar is bright.
		// The + 4 is because the text is indented 3 pixels and 1 extra pixel to the right as buffer.
		boolean textDark = endOffsX >= textWidth + 4 && this.colorIsBright;
		context.drawText(txtRend, bar, barX + 3, barY + 2, textDark ? 0xff000000 : 0xffffffff, !textDark);
```